### ISSUE TITLE
Reparenting RigidBody2D with physics interpolation enabled causes ghosting

### ISSUE THREAD

**Issue by blam23** [godotengine/godot#96663]
### Tested versions

- Reproducible in 4.3.stable

### System information

Godot v4.3.stable - Windows 10.0.22631 - GLES3 (Compatibility) - NVIDIA GeForce RTX 3080 Ti (NVIDIA; 31.0.15.5161) - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Issue description

When reparenting a node, and keeping it's _global_ position the same, the physics interpolation will cause a ghosting effect:

https://github.com/user-attachments/assets/ec97e25e-d4f6-474c-b2bc-2084ef747270

Here you can see when the parent node changes, there is a noticable ghosting - note that I have physics ticks per secondturned way down to 10 here to emphasise the issue.

I'm not sure if this is intentional or not - it can be addressed with `reset_physics_interpolation` (although that adds a bit of a stutter, might be due to very low physics tick rate) but I thought I'd report it here as it could be seen as unexpected as the body is technically in the same position, but with a different parent / local position.

[Documentation of `reset_physics_interpolation` states](https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-method-reset-physics-interpolation):
> When physics interpolation is active, moving a node to a radically different transform (such as placement within a level) can result in a visible glitch as the object is rendered moving from the old to new position over the physics tick.

I'm not sure I'd count the same global position as a "radically different transform" - I'd understand if the local transform changing radically is the cause of this flicker, I just think it should be explicitly stated.

### Steps to reproduce

- Enable physics interpolation
- Create a RigidBody2D and some kind of visual to track it
- Reparent the RigidBody2D while the game is running, keeping it's global position the same, such as:
```gdscript
func swap_parent(new_parent : Node2D):
	var pos := body.global_position
	body.get_parent().remove_child(body)
	
	new_parent.add_child(body)
	body.global_position = pos
```
- Body will visibly ghost.

### Minimal reproduction project (MRP)

4.3 project as per video:
[interpolation-reparent.zip](https://github.com/user-attachments/files/16911816/interpolation-reparent.zip)

**Comment by Sjoerd1988** [comment]
I'm having the same issue.
Calling ResetPhysicsInterpolation(); does not fix the issue for me:
- I've tried calling ResetPhysicsInterpolation **directly after** the reparenting, doesn't work.
- I've tried calling ResetPhysicsInterpolation **directly before** the reparenting, doesn't work, but this was to be expected.
- I've tried calling ResetPhysicsInterpolation **after waiting for 1 physics process frame**, also doesn't fix it.

**Comment by ghuser404** [comment]
I wonder why the engine doesn't do this automatically if it knows that upon changing position manually (or changing parent in this case) requires reseting physics interpolation. IMO, this is not something that the user should be bothered with.

**Comment by Sjoerd1988** [comment]
> I wonder why the engine doesn't do this automatically if it knows that upon changing position manually (or changing parent in this case) requires reseting physics interpolation. IMO, this is not something that the user should be bothered with.

Because the problem is the distance. When you reparent, the global position of that node becomes vec2(0, 0). Then it tries to interpolate to the newly set position, within a frame or 2. You still want the interpolation when you're offsetting to small distances.

**Comment by ghuser404** [comment]
> You still want the interpolation when you're offsetting to small distances.

I have a feeling that if you have a rigid body (which is controlled by physics), you want it to be controller by physics. But if you want to control it yourself, maybe you shouldn't be using rigid body. Is there an actuall use case when you want to offset by small distances and also have it as a rigid body?

**Comment by Nikitf777** [comment]
> maybe you shouldn't be using rigid body

If your game heavily relies on physics, you have to use rigid bodies for everything. Here's is a practical example of this issue:

https://github.com/user-attachments/assets/a6385f7b-fc09-451b-ad9a-2c02c4dee735

**Comment by blam23** [comment]
> > maybe you shouldn't be using rigid body
> 
> If your game heavily relies on physics, you have to use rigid bodies for everything. Here's is a practical example of this issue:
> physics-interpolation-ghosting.mp4

The issue in your video, I believe, is just a lack of a call to `reset_physics_interpolation` on the body. See: https://docs.godotengine.org/en/stable/tutorials/physics/interpolation/using_physics_interpolation.html#call-reset-physics-interpolation-when-teleporting-objects

You probably want to use the "moving start" method described in the link above.

> Moving start (e.g. bullet)
> 
>- Set the initial transform
> 
>- Call reset_physics_interpolation()
> 
>- Immediately set the transform expected after the first tick of motion
>

**Comment by Nikitf777** [comment]
Yes, it helps. The method should be called right after reparenting:
```gdscript
pulling_arrow.reparent(World.instance)
pulling_arrow.reset_physics_interpolation()
```
At this point, the correct solution might be to just document it better, because currently that page only mention teleporting, not reparenting.

**Comment by blam23** [comment]
> At this point, the correct solution might be to just document it better, because currently that page only mention teleporting, not reparenting.

I agree.
